CONVENTIONS.md

```diff
@@ -0,0 +1,55 @@
+
+"Duck interfaces" (Interfaces that are implemented onto classes using mixins) should be named the same name as the target class that implements the interface, with a series of adjectives at the front of the class name describing the contents or the implementation use case of the interface, along with "Hooks" at the end of the class name. For example, a duck interface implemented by `PlayerEntity` through mixins in order to add a mana player attribute should be named `ManaPlayerEntityHooks`.
+
+Mixins that shadow final fields must use the `@Final` annotation provided by the mixin library.
+
+Mixin injection and redirect methods must be named based upon the specific function of the injection or redirection, not the method that they are injecting into or redirecting. For example a method inject that registers commands on the dedicated server ought to be named `registerCommands` not the name of the target method, in this case `setupServer`.
+
+If a shadowed method needs to be prefixed because of a conflict it must use the prefix `minecraft$` for methods in minecraft, `[namespace]$` for projects with a different namespace (which should always match the mod id), or if neither of those are available `[package name]$`. Take for example a situation where the shadowed method `fromJson` conflicts with a new `fromJson` method. The shadowed method should be renamed to `minecraft$fromJson` and annotated as follows `@Shadow(prefix = "minecraft$")`.
+
+All mixin classes soft implementing interfaces must use the prefix `fabric$` and the `@Implements` annotation from the mixin library. For example, take a class that soft implements the interface `Identifiable`, it must be annotated as follows `@Implements(@Interface(iface = Identifiable.class, prefix = "fabric$"))`.
+
+## Project Structure
+
+Modules that do not contain API (API in data form is still considered API, such as tags) should not have an `-api` suffix.
+
+Event-based and registry-based modules must use the `-events` and `-registries` suffixes, respectively. These are API, however the `-api` suffix is redundant because events and registries are inherently API.
+
+Modules names will always be written in plain, grammatical, U.S. English. In English, adjectives come before nouns so, as an example, modules should be named `fabric-indigo-renderer` instead of such as `fabric-renderer-indigo`.
+
+Every name consisting of multiple words must have a hypen (-) between the words. This includes class names such as BlockEntity being represented as `block-entity`.
+
+Packages must follow the following format (using `fabric-biome-api-v1` as an example). API Packages must use the root `net.fabricmc.fabric.api.biome.v1`. Implementation Packages must use the root `net.fabricmc.fabric.impl.biome`. Mixin Packages must use the root `net.fabricmc.fabric.mixin.biome`.
```

Comment:
Client packages should be mentioned here as their standard is slightly different.